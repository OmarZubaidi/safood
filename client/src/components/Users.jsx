// Package imports
import React from 'react';
import { Card } from 'react-bootstrap';

// Local imports
import omar from '../img/omar.jpeg';

export default function Users ({ user }) {
  return (
    <Card
      className='text-center mb-4'
      style={{ width: '18rem' }}
    >
      <Card.Img src={omar} alt={user.img} />
      <Card.Header className='text-center mb-4 fs-3'>
        {user.name}
      </Card.Header>
      <Card.Title className='text-center'>
        About me
      </Card.Title>
      <Card.Body>
        {user.about}
      </Card.Body>
    </Card>
  );
}
