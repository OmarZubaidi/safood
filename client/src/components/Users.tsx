// Package imports
import React from 'react';
import { Card } from 'react-bootstrap';

// Local imports
import {IUserProps} from '../interfaces/User.interface'
import omar from '../img/omar.jpg';
import blueOmar from '../img/blueOmar.jpg';
import greenOmar from '../img/greenOmar.jpg';
import purpleOmar from '../img/purpleOmar.jpg';
import redOmar from '../img/redOmar.jpg';
import yellowOmar from '../img/yellowOmar.jpg';

export default function Users({ user }: IUserProps) {
  let img;
  switch (user.img) {
    case 'blue':
      img = blueOmar;
      break;
    case 'green':
      img = greenOmar;
      break;
    case 'purple':
      img = purpleOmar;
      break;
    case 'red':
      img = redOmar;
      break;
    case 'yellow':
      img = yellowOmar;
      break;
    default:
      img = omar;
  }

  return (
    <Card
      className='text-center mb-4'
      style={{ width: '18rem' }}
    >
      <Card.Img
        src={img}
        alt='User image'
      />
      <Card.Header className='text-center mb-4 fs-3'>
        {user.name}
      </Card.Header>
      <Card.Title className='text-center'>
        About me
      </Card.Title>
      <Card.Body>
        {user.about}
      </Card.Body>
    </Card>
  );
}
