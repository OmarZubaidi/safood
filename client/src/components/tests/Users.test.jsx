// Package imports
import { render, screen } from '@testing-library/react';

// Local imports
import Users from '../Users';

const mockUser = {
  name: 'blue',
  img: '',
  about: 'I\'m blue da ba dee da ba die!'
};

describe('Users component', () => {
  test('should match the snapshot', () => {
    const { container } = render(<Users user={mockUser} />);
    expect(container).toMatchSnapshot();
  });

  test('should render the user card', async () => {
    render(<Users user={mockUser} />);
    const userName = await screen.findByText(mockUser.name);
    const aboutTitle = await screen.findByText(/About me/i);
    const userAbout = await screen.findByText(mockUser.about);
    expect(userName.textContent).toBe(mockUser.name);
    expect(aboutTitle.textContent).toBe('About me');
    expect(userAbout.textContent).toBe(mockUser.about);
  });
});