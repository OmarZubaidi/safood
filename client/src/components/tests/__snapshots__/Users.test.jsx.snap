// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component should match the snapshot 1`] = `
<div>
  <div
    class="text-center mb-4 card"
    style="width: 18rem;"
  >
    <img
      alt="User image"
      class="card-img"
      src="omar.jpg"
    />
    <div
      class="text-center mb-4 fs-3 card-header"
    >
      blue
    </div>
    <div
      class="text-center card-title h5"
    >
      About me
    </div>
    <div
      class="card-body"
    >
      I'm blue da ba dee da ba die!
    </div>
  </div>
</div>
`;
